### PR TITLE
Update test_begin tests to synchronize tasks with sync variables

### PR DESCRIPTION
**Pull request by daviditen**
These tests violated the memory consistency model by updating an array in
one task and expecting to see the update in another task without any other
synchronization. Add synchronization to force the update to be visible.

### CODE REVIEW

**Comment by daviditen** [comment]
These tests were passing on master even though they're not entirely correct, but with an optimization I'm working on to replace array accesses with references they started to fail.  The task in the spin loop never saw the array updates from the other task and got stuck.

File: test/parallel/begin/deitz/test_begin.chpl [hunk 1]
```diff
@@ -3,12 +3,15 @@ use Time;
...
+var B: [1..n] sync bool;
 
 for i in 1..n do begin {
   a[i] = foo(i);
+  B[i] = true;
 }
 
 for i in 1..n {
+  B[i];
   while a[i] == 0 do
```
Comment (gbtitus) [inline test/parallel/begin/deitz/test_begin.chpl]: With the sync var there to control synchronization you should be able to get rid of the while-loop wait. You should be guaranteed that a(i)!=0 as soon as B(i) is non-empty (as long as foo(i) always returns non-0, of course).

File: test/parallel/begin/deitz/test_begin.chpl [hunk 2]
```diff
@@ -3,12 +3,15 @@ use Time;
 const n = 5;
 
 var a: [1..n] int;
+var B: [1..n] sync bool;
```
Comment (gbtitus) [inline test/parallel/begin/deitz/test_begin.chpl]: By convention (according to the spec) sync and single vars should end with '$' as an indication that references to them may block. So maybe rename to B$?

**Comment by gbtitus** [comment]
Looks good with a couple of minor changes, both of which apply to the 2nd file as well as the 1st.

**Comment by daviditen** [comment]
Thanks @gbtitus.  I changed the array-of-sync's name to 'B$' and replaced the wait loop with an assert.

**Pull request merged**